### ISSUE TITLE
make-wrapper.sh should fail if unknown option is given

### ISSUE THREAD

**Issue by ip1981** [NixOS/nixpkgs#3277]
make-wrapper.sh should fail if unknown option (e. g. with typos) is given.

**Comment by Profpatsch** [comment]
ack

```nix
with import <nixpkgs> {};

runCommand "foo" { buildInputs = [makeWrapper]; } ''
    touch $out
    wrapProgram $out --asdf bar
  ''
```

**Comment by Profpatsch** [comment]
There was #19061, but with problems on OSX. Maybe someone wants to give it another shot?

**Comment by jtojnar** [comment]
Fixed by #24944 and #25185
